promql/engine.go

```diff
@@ -648,19 +644,13 @@ func (ng *Engine) findMinTime(s *parser.EvalStmt) time.Time {
 	return s.Start.Add(-maxOffset)
 }
 
-func (ng *Engine) populateSeries(ctx context.Context, querier storage.Querier, s *parser.EvalStmt) (storage.Warnings, error) {
-	var (
-		// Whenever a MatrixSelector is evaluated, evalRange is set to the corresponding range.
-		// The evaluation of the VectorSelector inside then evaluates the given range and unsets
-		// the variable.
-		evalRange time.Duration
-		warnings  storage.Warnings
-		err       error
-	)
+func (ng *Engine) populateSeries(ctx context.Context, querier storage.Querier, s *parser.EvalStmt) {
```

Comment:
hm? :thinking: 

```suggestion
func (ng *Engine) asyncPopulateSeries(ctx context.Context, querier storage.Querier, s *parser.EvalStmt) {
```

Reply:
It doesn't necessarily need to be async. It depends on implementation. It puts an unexpanded SeriesSet to the statement node. I can still come up with a better name.

Reply:
Cool :+1: It's fine. Not a big deal, populate makes sense, just it does not mean things are populated when function finishes.